### ISSUE TITLE
Problems installing VEP (versions 95 and 95.1

### ISSUE THREAD

**Issue by CarlosGAH** [Ensembl/ensembl-vep#371]
Hi everybody
I am trying to install vep version 95 (and have tried with 95.1), and it fails. I am using Linux Ubuntu 18.04 and Perl v5.22.0.
Here is my command line and output
perl INSTALL.pl 

Hello! This installer is configured to install v95 of the Ensembl API for use by the VEP.
It will not affect any existing installations of the Ensembl API that you may have.

It will also download and install cache files from Ensembl's FTP server.

Checking for installed versions of the Ensembl API...done

Setting up directories
Destination directory ./Bio already exists.
Do you want to overwrite it (if updating VEP this is probably OK) (y/n)? y
 - fetching BioPerl
 - unpacking ./Bio/tmp/release-1-6-924.zip
 - moving files
Attempting to install Bio::DB::HTS and htslib.

>>> If this fails, try re-running with --NO_HTSLIB

 - checking out HTSLib
fatal: destination path 'htslib' already exists and is not an empty directory.
 - building HTSLIB in ./htslib
In /usr/local/ensembl-vep/htslib
make: Nothing to be done for 'all'.
 - unpacking ./Bio/tmp/biodbhts.zip to ./Bio/tmp/
./Bio/tmp/Bio-DB-HTS-2.9 - moving files to ./biodbhts
 - making Bio::DB:HTS
Created MYMETA.yml and MYMETA.json
Creating new 'Build' script for 'Bio-DB-HTS' version '2.9'
Building Bio-DB-HTS
.
.
.

Downloading required Ensembl API files
 - fetching ensembl
 - unpacking ./Bio/tmp/ensembl.zip
 - moving files
 - getting version information
 - fetching ensembl-variation
 - unpacking ./Bio/tmp/ensembl-variation.zip
 - moving files
 - getting version information
 - fetching ensembl-funcgen
 - unpacking ./Bio/tmp/ensembl-funcgen.zip
 - moving files
 - getting version information
 - fetching ensembl-io
 - unpacking ./Bio/tmp/ensembl-io.zip
 - moving files
 - getting version information

Testing VEP installation
./t/Parser_VEP_input.t ............................... Possible precedence issue with control flow operator at /usr/share/perl5/Bio/DB/IndexedBase.pm line 845.
./t/Parser_VEP_input.t ............................... 1/? 
   Failed test 'use Bio::EnsEMBL::VEP::Parser::VEP_input;'
   at ./t/Parser_VEP_input.t line 33.
     Tried to use 'Bio::EnsEMBL::VEP::Parser::VEP_input'.
     Error:  Base class package "Bio::EnsEMBL::IO::Parser" is not empty but "Bio/EnsEMBL/IO/Parser.pm" exists in the current directory.
     To help avoid security issues, base.pm now refuses to load optional modules
     from the current working directory when it is the last entry in @INC.
     If your software worked on previous versions of Perl, the best solution
     is to use FindBin to detect the path properly and to add that path to
    @INC.  As a last resort, you can re-enable looking in the current working
     directory by adding "use lib '.'" to your code.
  at Bio/EnsEMBL/IO/TextParser.pm line 45.
 BEGIN failed--compilation aborted at Bio/EnsEMBL/IO/TextParser.pm line 45.
 Compilation failed in require at /usr/share/perl/5.26/base.pm line 136.
 	...propagated at /usr/share/perl/5.26/base.pm line 158.
.
.
.
.
Parse errors: No plan found in TAP output
Files=47, Tests=923, 13 wallclock secs ( 0.11 usr  0.06 sys + 11.16 cusr  1.10 csys = 12.43 CPU)
Result: FAIL
Failed 33/47 test programs. 40/923 subtests failed.
The same happens with the  --NO_HTSLIB flag
[VEP_installation.txt](https://github.com/Ensembl/ensembl-vep/files/2803862/VEP_installation.txt)
I have uploaded the complete output. 
Thanks for your time and patience!

**Comment by dnebdal** [comment]
It looks like the "_To help avoid security issues, base.pm now refuses to load optional modules
from the current working directory when it is the last entry in @inc._" message is a response to https://bugzilla.redhat.com/show_bug.cgi?id=1355695 , FWIW.

(Oh, and "me too" on FreeBSD 12.)

**Comment by ens-lgil** [comment]
Dear @CarlosGAH,

It appears that you are actually using `Perl 5.26` and from this particular version the current directory is not included anymore in INC, as @dnebdal highlighted.

To fix this issue, you can add the path to your `ensembl-vep` into the `PERL5LIB` environment variable (you can add it in your `.bashrc` for instance), e.g.:
```
PERL5LIB=${PERL5LIB}::/path/to/ensembl-vep:/path/to/ensembl-vep/modules
export PERL5LIB
```
Best regards,
Laurent

**Comment by CarlosGAH** [comment]
Thanks for your responses. I have tried what you said Laurent, and still does not work. The same problems and output.
The strange thing is that the rest of the ensembl programs/aplications work perfectly, the only one that i am unable to install is the vep

**Comment by aparton** [comment]
Hi Carlos,

Sorry to hear that this has not resolved your issue. Without more information then it’s tricky to make a definitive diagnosis, but there are a couple of things we can look into.

1) Do you have multiple perl versions on your machine? You mentioned that you were running perl 5.22, however the logs are referring to /usr/share/perl/5.26/base.pm
2) Do you have an alternative version of Bio::DB::HTS installed on your machine? If this was installed for your previous Ensembl API installation, then version differences here could explain your failing tests
3) Could you please upgrade your perl Module::Build and try again? Should be as simple as 'cpanm Module::Build’ 

Kind Regards,
Andrew

**Comment by CarlosGAH** [comment]
Thanks Andrew, i have the last version of both modules:
Bio::DB::HTS is up to date. (2.11)
Module::Build is up to date. (0.4224)
and perl version: perl --version
This is perl 5, version 22, subversion 0 (v5.22.0) built for x86_64-linux-thread-multi
How can i see if i have more than one version of perl installed?
Thanks a lot for your time!

**Comment by ens-lgil** [comment]
Dear @CarlosGAH,
The default Perl version on Ubuntu 18.04 is perl 5.26 : [https://packages.ubuntu.com/bionic/perl](https://packages.ubuntu.com/bionic/perl).
To double check that you also have 5.26, you can look into the following directories:
```
/usr/share/perl/
/usr/local/share/perl/
/usr/local/bin/perl (check if it exists)
```
You can also try to run `which perl` and also `/usr/bin/perl -v`.

I hope this helps.

Best regards,
Laurent

**Comment by berguner** [comment]
Hello All,

I had a similar problem while I was trying to install vep on a Ubuntu 18.04 server. I was running the installer with sudo; `sudo perl INSTALL.pl` but it was not working because sudo did not propagate the PERL5LIB environment variable to the installer. Therefore I changed to root user with `sudo -s` and ran `export PERL5LIB=${PERL5LIB}:/path/to/ensembl-vep` to set the variable. Then I ran `perl INSTALL.pl` to install vep which worked without a problem.

I hope this solves your problems also.

Best,
Bekir
